pkg/asset/agent/image/ignitionbase.go

```diff
@@ -0,0 +1,303 @@
+	relImgData := struct{ ReleaseImage string }{
+		ReleaseImage: clusterImageSet.Spec.ReleaseImage,
+	}
+	for _, script := range []string{"release-image.sh", "release-image-download.sh"} {
+		if err := bootstrap.AddStorageFiles(&config,
+			"/usr/local/bin/"+script,
+			"bootstrap/files/usr/local/bin/"+script+".template",
+			relImgData); err != nil {
+			return err
+		}
+	}
+
+	err = addStaticNetworkConfig(&config, nmStateConfigs.StaticNetworkConfig)
+	if err != nil {
+		return err
+	}
+
+	// Enable pre-network-manager-config.service only when there are network configs defined
+	if len(nmStateConfigs.StaticNetworkConfig) != 0 {
+		agentEnabledServices = append(agentEnabledServices, "pre-network-manager-config.service")
+	}
+
+	filesToInclude := [...]asset.File{
+		*infraEnvAsset.File,
+		*clusterImageSetAsset.File,
+		*pullSecretAsset.File,
+	}
+
+	for _, file := range filesToInclude {
+		manifestFile := ignition.FileFromBytes(filepath.Join(manifestPath, filepath.Base(file.Filename)),
+			"root", 0600, file.Data)
+		config.Storage.Files = append(config.Storage.Files, manifestFile)
+	}
+
+	err = bootstrap.AddSystemdUnits(&config, "agent/systemd/units", agentTemplateData, agentEnabledServices)
+	if err != nil {
+		return err
+	}
+
+	addMirrorData(&config, registriesConfig, registryCABundle)
+
+	a.Config = &config
+
+	if err := a.generateFile(unconfiguredIgnitionFilename); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CopyIgnitionConfig returns a copy of the Ignition config
+// The copy is made by marshaling and unmarshaling the ignition yaml.
+func (a *IgnitionBase) CopyIgnitionConfig() (copy igntypes.Config, error error) {
+	copyData := &igntypes.Config{}
+	data, err := yaml.Marshal(a.Config)
+	if err != nil {
+		return copy, errors.Wrap(err, "copy failed, failed to Marshal IgnitionBase config")
+	}
+
+	if err := yaml.UnmarshalStrict(data, copyData); err != nil {
+		return copy, errors.Wrap(err, "copy failed, failed to Unmarshal IgnitionBase config")
+	}
+
+	return *copyData, nil
+}
+
+// PersistToFile writes the unconfigured ignition in the assets folder.
+func (a *IgnitionBase) PersistToFile(directory string) error {
+	if a.File == nil {
+		return errors.New("attempting to persist a IgnitionBase that has not been generated")
+	}
+	unconfiguredIgnFile := filepath.Join(directory, a.File.Filename)
+
+	err := os.WriteFile(unconfiguredIgnFile, a.File.Data, 0o644) //nolint:gosec // no sensitive info
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a *IgnitionBase) generateFile(filename string) error {
+	data, err := ignition.Marshal(a.Config)
+	if err != nil {
+		return errors.Wrap(err, "failed to Marshal Ignition config")
+	}
+	a.File = &asset.File{
+		Filename: filename,
+		Data:     data,
+	}
+	return nil
+}
+
+// Load returns the IgnitionBase from disk.
+func (a *IgnitionBase) Load(f asset.FileFetcher) (bool, error) {
+	// The IgnitionBase will not be needed by another asset so load is noop.
```

Comment:
We actually want to try using this asset from the appliance code, so could the func be implemented?

Reply:
This needs some explanation.

Reply:
Wanted to try importing the IgnitionBase asset in the appliance code. But nevermind, I understand it won't be supported. So we'll extract the installer image from the release and use the API.

Reply:
You want to shell out to the installer I think, otherwise you'll have to recompile your code every time you want to change the version to install.